### PR TITLE
DEV-3211: FPDS location data fix

### PR DESCRIPTION
**Pull request by alburde1**
**High level description:**
Creating a script to allow us to pull specific FPDS records for fixes.

**Technical details:**
Script takes a file with as many unique key columns as possible (preferably all of them). Also cleaning up the fpds loader and moving historic loading to a different file.

**Link to JIRA Ticket:**
[DEV-3211](https://federal-spending-transparency.atlassian.net/browse/DEV-3211)

The following are ALL required for the PR to be merged:
- [ ] Backend review completed
- Unit & integration tests updated with relevant test cases
- Frontend impact assessment completed

### CODE REVIEW

File: dataactcore/scripts/pull_fpds_data.py [hunk 1]
```diff
@@ -1502,7 +1500,7 @@ def get_data(contract_type, award_type, now, sess, sub_tier_list, county_by_name
             except KeyError:
                 continue
 
-            if last_run:
+            if last_run or specific_params:
```
Comment (dpb-bah) [inline dataactcore/scripts/pull_fpds_data.py]: I'd like to walk through the different workflows between `last_run`, `specific_params`, and everything else to make sure pairing `last_run` and `specific_params` is fit.

File: dataactcore/scripts/load_specific_fpds.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+        Args:
+            columns: The columns in the header of the provided file
+
+        Returns:
+            A dict containing all the FPDS query columns that the provided file has
+    """
+    possible_cols = {'agency_id': 'AGENCY_CODE:"{}" ', 'referenced_idv_agency_iden': 'REF_IDV_AGENCY_ID:"{}" ',
+                     'piid': 'PIID:"{}" ', 'award_modification_amendme': 'MODIFICATION_NUMBER:"{}" ',
+                     'parent_award_id': 'REF_IDV_PIID:"{}" ', 'transaction_number': 'TRANSACTION_NUMBER:"{}" ',
+                     'test_key': "This is a test"}
```
Comment (dpb-bah) [inline dataactcore/scripts/load_specific_fpds.py]: Are we keeping `test_key`?

File: dataactcore/scripts/load_specific_fpds.py [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+def get_param_cols(columns):
+    """ Get the columns that were provided in the file and return that list so we don't try to query non-existent cols
+
+        Args:
+            columns: The columns in the header of the provided file
+
+        Returns:
+            A dict containing all the FPDS query columns that the provided file has
+    """
+    possible_cols = {'agency_id': 'AGENCY_CODE:"{}" ', 'referenced_idv_agency_iden': 'REF_IDV_AGENCY_ID:"{}" ',
```
Comment (dpb-bah) [inline dataactcore/scripts/load_specific_fpds.py]: Totally nitpicky and can be disregarded. To reduce the extra `"{}"` in each of the values, would it make more sense to have `possible_cols` just store the columns like `{'agency_id': 'AGENCY_CODE', ...}` and have `create_param_string` handle the string formatting `'{}:"{}" '`? Again, very trivial.

**Comment by dpb-bah** [comment]
I'm assuming this has been tested locally. Are there any easy unit tests for this or run on sandbox?

**Comment by alburde1** [comment]
No easy unit tests. We can run it on sandbox if we want to wait until tomorrow to merge it (unless we want to do a small test set)

**Pull request merged**